Feat(html5): Add support for quizzes

Tainan404 opened the pull request:
### What does this PR do?
This PR introduces a new feature: quizzes.

It adds two new settings:

Quiz (boolean): Enables or disables quizzes on the server.

correctAnswerMarker (string): Defines the marker used to identify a quiz answer as correct in quick polls.

Due to a limitation in our PDF conversion process, trailing spaces at the end of lines are not preserved. This prevented us from implementing the initial idea of using double spaces as the marker for correct answers. Next week, we plan to investigate whether there’s a way to preserve trailing spaces in slides, so we can hopefully adopt " " as the marker in the future.


### Closes Issue(s)
Part of #23465


### More
Starting aquiz:

https://github.com/user-attachments/assets/0bb168f4-1a3b-4732-8f0b-90d7d4df6b2f




Using quick polls to start a quiz:

https://github.com/user-attachments/assets/4387f592-5761-4391-b940-4f1f2359a50d


Live results:

[Screencast from 04-07-2025 12:36:48.webm](https://github.com/user-attachments/assets/112a7e74-a240-4e85-b76e-e5fd50b60e1b)

gustavotrott committed:
backend (feat): Add support for polls with correct answers (quizzes)

gustavotrott committed:
Temporarily disable the removal of unpublished polls from the Dashboard while the appropriate handling of this case is still under internal discussion

Tainan404 committed:
WIP(html5): Quiz frontend

Tainan404 committed:
Feat(html5): Add support for quizzes

Tainan404 committed:
Add missing locale

Tainan404 committed:
Add locales for quiz options

Tainan404 committed:
Add tooltip and change to radio

gustavotrott committed:
provide poll correctAnswer through Graphql

Tainan404 committed:
fix lint

gustavotrott committed:
document: provide poll correctAnswer through Graphql

gustavotrott committed:
fix client with missing props for createPoll mutation

Tainan404 committed:
Fix typo

Tainan404 committed:
Merge branch 'PR_23488' into quiz-implementation

Tainan404 committed:
Add live result view

Tainan404 committed:
Add verification to ensure a correct answer at quiz

Tainan404 committed:
Make Show quiz result comes marked

Tainan404 committed:
Add accessibility for live results correct answer

Tainan404 committed:
Merge remote-tracking branch 'upstream/v3.0.x-release' into quiz-implementation

gustavotrott commented on bigbluebutton-html5/private/config/settings.yml:
```diff
@@ -584,6 +584,8 @@ public:
     chatMessage: true
     quickPollConfirmationStep: true
+    quiz: true
+    correctAnswerMarker: "__"
```
I suggest make it clear that `correctAnswerMarker` is related with "quick poll" feature

```suggestion
    quiz:
      enabled: true
      quickPollCorrectAnswerSuffix: "__"
```

Tainan404 committed:
Fix cancel poll break quiz correct answer verification

Tainan404 committed:
Reset correct answer when poll type changes

Tainan404 committed:
Change locale string

Tainan404 committed:
Fix wrong title when quizzing with question

Tainan404 committed:
Fix eslint

gustavotrott committed:
Add new prop for Poll chat message metadata `isCorrectAnswer`
Stop including GreenCheck for poll options labels, as it mess the translations
Include the GreenCheck for correct answer only for whiteboard (temporary)

gustavotrott commented:
https://github.com/Tainan404/bigbluebutton/pull/37

gustavotrott commented:
The correct answer should not be translated!

![image](https://github.com/user-attachments/assets/0b1b56a1-b6f1-4e71-b696-3a27135670f3)

Tainan404 committed:
Merge pull request #37 from gustavotrott/pr-23506

Tweak backend side of Polls to make translations work on client side

Tainan404 committed:
Fix translated responses

Tainan404 committed:
Merge branch 'quiz-implementation' of github.com:Tainan404/bigbluebutton into quiz-implementation

gustavotrott commented:
I'm not being able to start a normal poll (not quiz) using Smart slide:


https://github.com/user-attachments/assets/eba27b67-d2bd-4d2a-a440-72a25407528b

The file I'm using to test:
[teste.odp](https://github.com/user-attachments/files/21066059/teste.odp)

Tainan404 committed:
Fix quick polls not working

Tainan404 committed:
fix lint

Tainan404 committed:
Add quiz title to chat response

Tainan404 committed:
Fix quick poll dropdown error

gustavotrott commented on bigbluebutton-html5/public/locales/en.json:
```diff
@@ -440,6 +441,17 @@
+    "app.poll.quiz": "Quiz",
+    "app.poll.quiz.error": "Quiz has not correct answers selected. Please select at least one correct answer.",
+    "app.poll.quiz.start.label": "Start Quiz",
+    "app.poll.quiz.showCorrectAnswer": "Show correct answer to all users",
```
what do you think?
```suggestion
    "app.poll.quiz.showCorrectAnswer": "Display correct answer to all participants",
```

gustavotrott commented on bigbluebutton-html5/public/locales/en.json:
```diff
@@ -440,6 +441,17 @@
+    "app.poll.quiz.start.label": "Start Quiz",
+    "app.poll.quiz.showCorrectAnswer": "Show correct answer to all users",
+    "app.poll.quiz.options.title": "Quiz options",
+    "app.poll.quiz.options.tooltip": "Select this answer as the correct one",
```
```suggestion
    "app.poll.quiz.options.tooltip": "Set as correct answer",
```

gustavotrott replied:
And the ID `app.poll.quiz.options.tooltip` is vague about the specific element it represents.

gustavotrott commented on bigbluebutton-html5/public/locales/en.json:
```diff
@@ -440,6 +441,17 @@
+    "app.poll.quiz.options.correct": "Correct Answer",
+    "app.poll.quiz.options.incorrect": "Incorrect Answer",
+    "app.poll.quiz.liveResult.title.correct": "Correct",
+    "app.poll.quiz.options.responseNotSecret": "Quiz – the presenter can see your answer.",
```
Is the final period in this translation intentional? as it doesn’t seem to be the standard to include one

Tainan404 replied:
I just followed the same text of the polls 
![image](https://github.com/user-attachments/assets/c052b9b6-cf92-4a3a-b2e4-f0156a74985f)

Tainan404 committed:
Update bigbluebutton-html5/public/locales/en.json

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

Tainan404 committed:
Update bigbluebutton-html5/public/locales/en.json

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

Tainan404 committed:
Add translations to accebility tags

Tainan404 committed:
Merge branch 'quiz-implementation' of github.com:Tainan404/bigbluebutton into quiz-implementation

Tainan404 committed:
Make quiz disable by default

Tainan404 committed:
rename settings field

Tainan404 committed:
Rename local variable to reflect settings name

Tainan404 committed:
add a new level for quiz on settings

Tainan404 committed:
Fix ts error
